### ISSUE TITLE
Loss curve during training looks unstable

### ISSUE THREAD

**Issue by daichi1207** [md-mohaiminul/BIMBA#4]
Hello, thank you for the great work!

I have been trying to reproduce the results using the exp_small.yaml configuration, but I’m seeing a strange loss curve compared to what I expected. The loss fluctuates heavily and does not show a clear downward trend.

I've pasted a snapshot of my training logs (W&B plots) and train loss fluctuates between ~0.2–0.6 without a clear decrease.

<img width="1390" height="669" alt="Image" src="https://github.com/user-attachments/assets/8dcdfd54-fdfc-4501-ac4b-718493744832" />

Below is the script I used:

```
VISION_MODEL_VERSION="google/siglip-so400m-patch14-384"
PROMPT_VERSION="qwen_1_5"
PREV_STAGE_CHECKPOINT="lmms-lab/LLaVA-Video-7B-Qwen2"
RUN_NAME="BIMBA_LLaVA_Qwen2_7B_"
OUTPUT_DIR="work_dirs/${RUN_NAME}"

export WANDB_PROJECT=BIMBA

deepspeed --master_port 30000 \
    llava/train/train_mem.py \
    --deepspeed scripts/zero3.json \
    --model_name_or_path $PREV_STAGE_CHECKPOINT \
    --lora_enable True \
    --lora_r 128 \
    --lora_alpha 32 \
    --version $PROMPT_VERSION \
    --data_path $DATA_YAML \
    --image_folder $IMAGE_FOLDER \
    --video_folder $VIDEO_FOLDER \
    --mm_tunable_parts="mm_vision_tower,mm_mlp_adapter,mm_language_model" \
    --mm_vision_tower_lr=2e-6 \
    --vision_tower ${VISION_MODEL_VERSION} \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --compressor_type bimba \
    --group_by_modality_length True \
    --image_aspect_ratio anyres_max_9 \
    --image_grid_pinpoints  "(1x1),...,(6x6)" \
    --mm_patch_merge_type spatial_unpad \
    --bf16 True \
    --run_name $RUN_NAME \
    --output_dir $OUTPUT_DIR \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 500 \
    --save_total_limit 100 \
    --learning_rate 1e-6 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 32768 \
    --gradient_checkpointing True \
    --dataloader_num_workers 2 \
    --lazy_preprocess True \
    --torch_compile True \
    --torch_compile_backend "inductor" \
    --dataloader_drop_last True \
    --frames_upbound 64 \
    --mm_newline_position grid \
    --add_time_instruction True \
    --force_sample True \
    --mm_spatial_pool_stride 2 \
    --report_to wandb
```

Did you observe a similar unstable loss curve when training with this setup, or should I expect the loss to decrease more smoothly?
I would like to confirm whether this behavior is normal for the exp_small.yaml configuration, or if I may have misconfigured something.

Thanks again for your help!
